docs/entity-manager-api.md

```diff
@@ -147,6 +147,29 @@ await manager.update(User, 1, { firstName: "Rizzrak" });
 // executes UPDATE user SET firstName = Rizzrak WHERE id = 1
 ```
 
+* `upsert` - Inserts a new entity or array of entities unless they already exist in which case they are updated instead. Supported by AuroraDataApi, Cockroach, Mysql, Postgres, and Sqlite database drivers.
+
+```typescript
+await manager.upsert(User, { externalId: "abc123" }, { firstName: "Rizzrak" });
```

Comment:
TBH I find this signature confusing. I prefer to have only second signature:

```ts
await manager.update(User, ["externalId"], [
    { externalId:"abc123", firstName: "Rizzrak" },
    { externalId:"bca321", firstName: "Karzzir" },
    ]);
```

Reply:
Also I would swap parameters order, it makes more logically sense to have:

```ts
await manager.update(
  User, 
  [
      { externalId:"abc123", firstName: "Rizzrak" },
      { externalId:"bca321", firstName: "Karzzir" },
  ], 
  ["externalId"]
);
```

Reply:
I put a lot of thought into the signature and think it is best to match the style used by https://docs.mongodb.com/manual/reference/method/db.collection.findOneAndUpdate/. In addition, since the definition is essentially `upsert(Entity, ${things to match/conflict on}, ${things to update})` I felt that parameter order made more sense. I feel strongly about keeping the first signature, but a little less strongly about the order of the second signature.

Reply:
`${things to match/conflict on}` - but when there is no conflict, this value will be used for insertion, right? So it's actually `${things to match/conflict on/insert if no conflict merged with next parameter values}` - and this is what it makes this signature complicated in understanding.

Reply:
this is where i'm hung up:

`await postRepository.upsert({ externalId }, { title: "Post title initial" });`

vs

`await postRepository.upsert({ externalId, title: "Post title initial" }, ["externalId"]);`

I feel that the first signature flows significantly better than the second. The second feels unnecessarily redundant. `MongoRepository#updateOne` has the same basic signature and I don't believe it's complicated to understand: `await mongoRepository.updateOne({ externalId }, { title: "Post title initial" }, { upsert: true });`

Perhaps there's a middle ground here? Maybe rather than 1 upsert method with 2 signatures I should introduce 2 upsert methods? `Repository#upsertOne(conditions, partial)`, a signature similar to `MongoRepository#updateOne(conditions, partial, { upsert: true })` and also `Repository#upsertMany(conflictPaths: string[], partialEntities: DeepPartial<Entity>[])`

I really don't want to lose the MongoDb style signature.

Reply:
I'm looking for both simplicity and obviousness. Adding new method won't help. Following MongoDB convention doesn't have to be there, maybe some people can appreciate it, but most prefer simplicity and obviousness. And I don't find `${things to match/conflict on/insert if no conflict merged with next parameter values}` parameter obvious. Maybe because I'm not MongoDB user. I would prefer to type extra code.

Reply:
Well, I don't like it, but you're in charge here. I'll update the PR with your preferred style.

Reply:
sorry about that =( If you want we can wait for other people who are interested in this feature to provide their feedback.

Reply:
no worries, I'm going to make the change. we can always reintroduce the second signature if feedback warrants it. thanks for everything you do for typeorm.

Reply:
thank you 🤞